app/controllers/commits_controller.rb

```diff
@@ -63,16 +64,36 @@ def show
 
 private
 
+  def reset_params
+    if !params[:status_type_id] && !params[:direction] && !params[:order]
+      session.delete(:direction)
+      session.delete(:order)
+      session.delete(:status_type_id)
+    end
+  end
+
+  def set_params
+    if params[:status_type_id]
```

Comment:
Not a merge block but it's more idiomatic to use single line ifs here https://github.com/bbatsov/ruby-style-guide#if-as-a-modifier